src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/RoutePatternAnalyzer.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.RoutePattern;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class RoutePatternAnalyzer : DiagnosticAnalyzer
+{
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(new[]
+    {
+        DiagnosticDescriptors.RoutePatternIssue
+    });
+
+    public void Analyze(SemanticModelAnalysisContext context)
+    {
+        var semanticModel = context.SemanticModel;
+        var syntaxTree = semanticModel.SyntaxTree;
+        var cancellationToken = context.CancellationToken;
+
+        var root = syntaxTree.GetRoot(cancellationToken);
+        Analyze(context, root, cancellationToken);
+    }
+
+    private void Analyze(
+        SemanticModelAnalysisContext context,
+        SyntaxNode node,
+        CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        foreach (var child in node.ChildNodesAndTokens())
+        {
+            if (child.IsNode)
+            {
+                Analyze(context, child.AsNode()!, cancellationToken);
+            }
+            else
+            {
+                var token = child.AsToken();
+                if (!RouteStringSyntaxDetector.IsRouteStringSyntaxToken(token, context.SemanticModel, cancellationToken))
```

Comment:
so my preference would be that roslyn expose an embedded lang analyzer, so you don't have to do all this work and duplciation.